shared/core/types/pools_test.go

```diff
@@ -0,0 +1,67 @@
+package types
+
+import (
+	"bytes"
+	"encoding/hex"
+	"errors"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestPools_Address(t *testing.T) {
+	tests := []struct {
+		name string
+		pn   Pools
+	}{
+		// initializing tests with the custom/edge cases
+		{"unspecified", Pools_POOLS_UNSPECIFIED},
+		{"invalid", Pools(100)},
+	}
+
+	// adding all the real world cases programmatically in order to catch any changes to the enum
+	// that must be reflected into the hardcoded values
+	for _, pool := range Pools_value {
+		tests = append(tests, struct {
+			name string
+			pn   Pools
+		}{
+			name: Pools(pool).FriendlyName(),
+			pn:   Pools(pool),
+		})
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			name := tt.pn.FriendlyName()
+			want, err := convertFriendlyNameToHexBytes(name)
+			require.NoError(t, err)
+
+			if got := tt.pn.Address(); !bytes.Equal(got, want) {
+				t.Errorf("Pools.Address() = %v, want %v", string(got), string(want))
+			}
+		})
+	}
+}
+
+// convertFriendlyNameToHexBytes is the function used to opinionatedly convert a pool name into a valid address
```

Comment:
Thoughts on doing the same thing as v0? https://github.com/pokt-network/pocket-core/blob/a109dfc03a13eec06413bf1eb7d17fe093f96842/x/auth/types/account.go#L320

Leaving a TODO is fine a well - up to you.

Reply:
I would but 🦶🚪and I can foresee a long night.
Adding a TODO